File: python/examples/example-module-actor.py
# SPDX-License-Identifier: BSL-1.0

import sys
import time
import traeger

from example_actor_messaging import perform_operations


def main(argv: list[str]):
    if len(argv) < 2:
        print("Missing argument")
        return 1

    path: str = argv[1]
    print(f"Attempting to load module from path: {path}")
    configuration: traeger.Map = traeger.Map({"initial_funds": 100.0})
    module: traeger.Module = traeger.Module(path, configuration)
    scheduler: traeger.Scheduler = traeger.Scheduler(threads_count=8)
    perform_operations(scheduler, module.mailbox())

    while scheduler.count() != 0:
        time.sleep(0.010)
    return 0


if __name__ == "__main__":
    sys.exit(main(sys.argv))

File: rust/examples/example-module-actor.rs
// SPDX-License-Identifier: BSL-1.0

use std::{env, thread::sleep, time::Duration};

use traeger::{Module, Scheduler, Threads, map};

pub mod example_actor_messaging;
use example_actor_messaging::perform_operations;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        println!("Missing argument");
        return;
    }

    let path = &args[1];
    println!("Attempting to load module from path: {path}");

    match Module::from_path(&path, &map!("initial_funds" => 1000.0)) {
        Err(error) => {
            println!("Module error: {error}");
        }
        Ok(module) => {
            let scheduler = Scheduler::new(Threads { count: 8 });
            let mailbox = module.mailbox();
            perform_operations(&scheduler, &mailbox);

            while scheduler.count() != 0 {
                sleep(Duration::from_millis(10));
            }
        }
    }
}
